examples/transform/src/test/java/org/apache/apex/examples/transform/SimpleTransformApplicationTest.java

```diff
@@ -0,0 +1,95 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.apex.examples.transform;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import org.codehaus.jackson.map.ObjectMapper;
+import org.hamcrest.Matchers;
+import org.junit.Test;
+
+import org.apache.apex.api.EmbeddedAppLauncher;
+import org.apache.apex.api.Launcher;
+import org.apache.apex.malhar.lib.function.Function;
+import org.apache.hadoop.conf.Configuration;
+
+import com.datatorrent.api.Attribute;
+
+import static junit.framework.Assert.assertEquals;
+import static junit.framework.Assert.assertFalse;
+import static org.junit.Assert.assertThat;
+
+public class SimpleTransformApplicationTest
+{
+  private static final List<Object> results = Collections.synchronizedList(new ArrayList<>());
+
+  @SuppressWarnings("serial")
+  private static final Function.MapFunction<Object, Void> outputFn = new Function.MapFunction<Object, Void>()
+  {
+    @Override
+    public Void f(Object input)
+    {
+      results.add(input);
+      return null;
+    }
+  };
+
+  @Test
+  public void testApplication() throws Exception
+  {
+    Configuration conf = new Configuration(false);
+    conf.addResource(this.getClass().getResourceAsStream("/META-INF/properties.xml"));
+    EmbeddedAppLauncher<?> launcher = Launcher.getLauncher(Launcher.LaunchMode.EMBEDDED);
+    Attribute.AttributeMap launchAttributes = new Attribute.AttributeMap.DefaultAttributeMap();
+    launchAttributes.put(EmbeddedAppLauncher.RUN_ASYNC, true);
+    SimpleTransformApplication simpleTransformApplication = new SimpleTransformApplication();
+    simpleTransformApplication.outputFn = outputFn;
+    Launcher.AppHandle appHandle = launcher.launchApp(simpleTransformApplication, conf, launchAttributes);
+    int sleepTimeCounterForLoopExit = 0;
+    int sleepTimePerIteration = 500;
+    // wait until expected result count or timeout
+    while (results.size() < simpleTransformApplication.getPojoDataGenerator().getMaxTuples()) {
+      sleepTimeCounterForLoopExit += sleepTimePerIteration;
+      if (sleepTimeCounterForLoopExit > 30000) {
+        break;
+      }
+      Thread.sleep(sleepTimePerIteration);
+    }
+    appHandle.shutdown(Launcher.ShutdownMode.KILL);
+    assertEquals(results.size(), simpleTransformApplication.getPojoDataGenerator().getMaxTuples());
+    assertTransformationsGenerated(simpleTransformApplication);
+  }
+
+  private void assertTransformationsGenerated(SimpleTransformApplication simpleTransformApplication) throws IOException
+  {
+    ObjectMapper mapper = new ObjectMapper();
+    POJOGenerator pojoGenerator = simpleTransformApplication.getPojoDataGenerator();
+    int maxLengthOfCustomerName = 1 + ( 2 * pojoGenerator.getMaxNameLength()); //first name + last name + space
+    for (Object stringJsonObject : results) {
```

Comment:
stringJsonObject is already CustomerInfo, right? So need need for toString and then parse again.

Reply:
fixed.